xds/internal/balancer/ringhash/weight_attribute.go

```diff
@@ -0,0 +1,40 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package ringhash
+
+import "google.golang.org/grpc/resolver"
+
+// weightAttributeKey is used as the attribute key when storing the address
+// weight in the `Attributes` field of the address.
+type weightAttributeKey struct{}
+
+// setWeightAttribute returns a copy of addr in which the Attributes field is
+// updated with weight.
+func setWeightAttribute(addr resolver.Address, weight uint32) resolver.Address {
+	addr.Attributes = addr.Attributes.WithValue(weightAttributeKey{}, weight)
```

Comment:
`BalancerAttributes`?

Hold up, are we saying we want an `AddressMap` that is sensitive to `BalancerAttributes` (or at least this one), i.e. attributes that do not impact subchannel uniqueness?  Because that's not what `AddressMap` is, currently.

Reply:
I did not change the logic here wrt to what was being used from `Attributes` and from `BalancerAttributes`. And I sort of get the idea that the `ringhash` LB policy cares only about address weight when deciding whether it needs to recreate subConns.

But I also seem to get your point that it doesn't seem to be right to be ignoring `Attributes` while deciding on subchannel uniqueness. Let me check what the other implementations do. Thanks.